scripts/run_tracker.py

```diff
@@ -172,6 +178,156 @@ def trigger_analysis(run, config):
+        the sample sheet created by GenoLogics in order to correctly demultiplex HiSeqX runs.
+        This function returns with success if the samplesheet is in the correct place, otherwise
+        this flowcell will not be processed.
+        
+        :param str run: Run directory
+        :param dict config: Parset configuration file
+        
+    """
+    #start by checking if samplesheet is in the correct place
+    run_name     = os.path.basename(run)
+    current_year = '20' + run_name[0:2]
+    samplesheets_dirs = config['samplesheets_dir']
+    samplesheets_dir  = os.path.join(samplesheets_dirs, current_year)
+
+    run_name_componets = run_name.split("_")
+    FCID = run_name_componets[3][1:]
+    
+    FCID_samplesheet_origin = os.path.join(samplesheets_dir, FCID + '.csv')
+    FCID_samplesheet_dest   = os.path.join(run, "SampleSheet.csv")
+
+    #check that the samplesheet is not already present
+    if os.path.exists(FCID_samplesheet_dest):
+        LOG.warn(("When trying to generate SampleSheet.csv for sample sheet {} I dinf out that "
+                 "SampleSheet.csv was already present in {} !!".format(FCID, FCID_samplesheet_dest)))
+        return False
+
+    FCID_samplesheet_origin_dict = sampleSheetToDict(FCID_samplesheet_origin) # store samplesheet in a dict
+
+    FCID_samplesheet_dest_dict = sampleSheetMassage(FCID_samplesheet_origin_dict) # massage the sample sheet
+
+    if not FCID_samplesheet_dest_dict: # something went wrong when working on samplesheet
+        return False
+
+    dictToSampleSheet(FCID_samplesheet_dest_dict, FCID_samplesheet_dest) # store the new smaplesheet file in the fc dir
+
+    return True #everything ended corretly
+
+
+def sampleSheetMassage(samplesheet_dict):
+    """ ASSUMPTION: this samplesheet has been generated by genologics and is problematic.
+        it contains index2 in Data section than needs to be removed, and SampleID needs
+        to became SampleName
+        
+        :param dict FCID_samplesheet_origin_dict: the sample sheet stored in a hash table
+    """
+    
+    #check that sample sheet is ok
+    if not samplesheet_dict["Header"]:
+        LOG.warn(("When trying to generate SampleSheet.csv for sample sheet {} I find out that "
+          " it does not contain the Header section !!".format(FCID)))
+        return False
+
+    if not samplesheet_dict["Data"]:
+        LOG.warn(("When trying to generate SampleSheet.csv for sample sheet {} I find out that "
+                  " it does not contain the Data section !!".format(FCID)))
+        return False
+
+
+    FCID_samplesheet_dest_dict = {}
+    FCID_samplesheet_dest_dict["Header"] = []
+    for header in samplesheet_dict["Header"]:
+        FCID_samplesheet_dest_dict["Header"].append(header)
+
+
+    FCID_samplesheet_dest_dict["Data"] = []
+    for data in samplesheet_dict["Data"]:
+        if data: #if not empty
+            new_data = data[0:6] + data[7:8] # remove index2 fuck Genologics
```

Comment:
are we sure these will always be the columns? well I guess we are not... but what else